docs/rfc/active/graph_ql_frontend/mutations.md

```diff
@@ -0,0 +1,647 @@
+For reference, check out `LogInResolver.java` included in the PoC PR. 
+
+**Authenticating Requests**
+
+To authenticate requests from a signed-in user, we introduce an abstract base GraphQL resolver which is responsible for validating the current client's access token, in turn depending on the `QueryContext` object which internally stores the Play-provided "session" object. Any GraphQL resolver that needs to be behind an authentication wall should extend this base class, overriding the `authenticatedGet` method. 
+
+For reference, check out `AuthenticatedResolver.java` included in the PoC PR.
+
+#### Search 
+
+There are 2 pieces of functionality comprising search:
+
+1. Full-text search against DH entities 
+2. AutoComplete of partial search text
+
+**Full-text Search**
+
+Full text search allows a user to query for entities of a particular type by a search term. To implement full-text search, we introduce the `search` query 
+
+```graphql
+type Query {
+    ...
+    search(input: SearchInput!): SearchResults
+}
+```
+
+which accepts a`SearchInput` 
+
+```graphql
+input SearchInput {
+	type: EntityType!
+	query: String!
+	start: Int
+	count: Int
+	filters: [FacetFilterInput]
+}
+```
+
+and returns `SearchResults` 
+
+```graphql
+type SearchResults {
+    start: Int!
+    count: Int!
+    total: Int!
+    elements: [SearchResult]!
+    facets: [FacetMetadata]
+}
+
+union SearchResult = Dataset | CorpUser
+```
+
+Behind the scenes, we implement a resolver which executes the downstream search against GMA and fits the returned data to the GQL schema.
+
+For reference, check out `SearchResolver.java` included in the PoC PR.
+
+**Auto Complete**
+
+Auto complete involves providing suggestions as a user enters types in the search box. To implement auto-complete, we introduce an `autoComplete` query
+
+```graphql
+type Query {
+    ....
+    autoComplete(input: AutoCompleteInput!): AutoCompleteResults
+}
+```
+which accepts an `AutoCompleteInput`
+
+```graphql
+input AutoCompleteInput {
+	type: EntityType!
+	query: String!
+	field: String! # Field name
+	limit: Int
+	filters: [FacetFilterInput]
+}
+```
+
+and returns `AutoCompelteResults`
+
+```graphql
+type AutoCompleteResults {toCom
+	query: String!
+	suggestions: [String]!
+}
+```
+
+We also introduce a resolver for autocomplete. For reference, check out `AutoCompleteResolver.java` included in the PoC PR.
+
+#### Browse
+
+There are 2 pieces of functionality comprising the browse experience:
+
+1. Browsing lists of entities by navigating through a tree of paths
+2. Retrieving the list of full paths corresponding to a particular entity
+
+**Browsing Entities**
+
+Browsing entities requires the ability to retrieve all groups or entities under a particular path. To achieve, this we introduce a `browse` query 
+
+```graphql
+type Query {
+    ...
+    browse(input: BrowseInput!): BrowseResults
+}
+```
+
+which accepts a`BrowseInput`
+
+```graphql
+input BrowseInput {
+	type: EntityType!
+	path: String
```

Comment:
Would this make more sense as `pathParts: String[]` so that encoding of paths is not needed?

Reply:
This should be better :)

Reply:
Any particular reason why?

Reply:
I'm generally not a fan of including string fields that are up for interpretation. Requires encoders and decoders, and often people don't make those but just do it in line and then the logic isn't reusable and is unclear and meaning gets lost to time. /rant

tl;dr avoid stringly typed code if possible